src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMaterializer.cs

```diff
@@ -36,12 +36,25 @@ public override JsonClassInfo.ConstructorDelegate CreateConstructor(Type type)
             return (JsonClassInfo.ConstructorDelegate)dynamicMethod.CreateDelegate(typeof(JsonClassInfo.ConstructorDelegate));
         }
 
-        public override object ImmutableCreateRange(Type constructingType, Type elementType)
+        public override object ImmutableCreateRange(Type constructingType, Type elementType, bool constructingTypeIsDict)
```

Comment:
Similar feedback about `bool isDictionary`. If you think it makes more sense, refactor this to be separate methods.

Reply:
Refactored.